packages/template-retail-react-app/app/pages/checkout/partials/shipping-multi-address.jsx

```diff
@@ -378,16 +511,68 @@ const ShippingMultiAddress = ({
                                             </Box>
                                         </VStack>
                                     </Flex>
+
+                                    {/* Add New Address Form - appears inside the product card */}
+                                    {showAddAddressForm[addressKey] && (
+                                        <Box position="relative" mt={4} width="100%">
+                                            <AddressForm
+                                                item={item}
+                                                form={addressForm}
+                                                onSubmit={handleCreateAddress}
+                                                onCancel={() => {
+                                                    setShowAddAddressForm((prev) => ({
+                                                        ...prev,
+                                                        [addressKey]: false
+                                                    }))
+                                                    // Clear form errors using react-hook-form
+                                                    addressForm.clearErrors()
+                                                }}
+                                            />
+                                        </Box>
+                                    )}
                                 </Box>
                             )
                         })}
                     </VStack>
                 </Box>
-                <Box pt={2} w="100%">
-                    <Button type="button" width="full" onClick={onSubmit}>
-                        {formatMessage(submitButtonLabel)}
-                    </Button>
-                </Box>
+                <Button
+                    type="button"
```

Comment:
This should be submit type, so we don't need to manually submit
```suggestion
                    type="submit"
```

Reply:
@alexvuong `type="submit"` will not work since there is no `<form>` tag around this button.